apps/vx-scan/frontend/src/screens/election_manager_screen.tsx

```diff
@@ -49,12 +49,12 @@ export function ElectionManagerScreen({
   isTestMode,
```

Comment:
While I see that components are now consistently pulling react-query mutations themselves, how have you been thinking about react-query data? Specifically when to pull it in directly vs. when to pass it through props or context? I know passing data through props can let you ensure that the data is loaded, avoiding unnecessary is-loading checks

Reply:
I have been leaning a bit towards pushing down queries and mutations as low in the component hierarchy as possible for a few reasons:
- Colocation makes it easier to read, especially for simple read/write components that update a single setting - you don't have to trace the source of the data back up the component hierarchy.
- Duplicated data loading in subcomponents is not a problem due to caching.
- It's nice when writing a component-level test to have the data loading and updating both happen within the component so you can easily simulate a full read/write/re-read cycle. This encourages writing tests for smaller components rather than the app-level tests we currently have (many of which just test one part of a specific sub-component, but are set up at the app level because that's where all the plumbing used to live)

However, this is explicitly the opposite of the general React guidance of lifting up shared state. So I'm not 100% convinced it's the way to go, but it's the way I've been leaning lately.

Reply:
We discussed and decided that it's usually good to write test that capture the entire scope of a feature, so locating the query for data used in a feature at the same level in the component hierarchy as the feature makes sense. We also should try not to write too many tests for subcomponents in order to give ourselves more flexibility about how we organize the component hierarchy over time.

This reminds me of the idea that integration tests are generally the best bang for your buck from this article: https://kentcdodds.com/blog/write-tests#mostly-integration
>Integration tests strike a great balance on the trade-offs between confidence and speed/expense.

Reply:
Will incorporate in follow up PR to remove app context